numpyro/distributions/util.py

```diff
@@ -282,6 +285,42 @@ def vec_to_tril_matrix(t, diagonal=0):
     return np.reshape(x, x.shape[:-1] + (n, n))
 
 
+def cholesky_update(L, x, coef=1):
+    """
+    Finds cholesky of L @ L.T + coef * x @ x.T.
+
+    **References;**
+
+        1. A more efficient rank-one covariance matrix update for evolution strategies,
+           Oswin Krause and Christian Igel
+    """
+    batch_shape = lax.broadcast_shapes(L.shape[:-2], x.shape[:-1])
+    L = np.broadcast_to(L, batch_shape + L.shape[-2:])
+    x = np.broadcast_to(x, batch_shape + x.shape[-1:])
+    diag = np.diagonal(L, axis1=-2, axis2=-1)
+    # convert to unit diagonal triangular matrix: L @ D @ T.t
+    L = L / diag[..., None, :]
```

Comment:
The implementation is a simplified version of Algorithm 3.1 of Ref [1] (here entries of L are divided by diagonal part by default).